tests/test_components/test_component.py

```diff
@@ -0,0 +1,47 @@
+
+# pylint: disable=protected-access, unused-argument
+# pylint: disable=no-value-for-parameter
+
+__copyright__ = "Copyright 2020, http://radical.rutgers.edu"
+__license__   = "MIT"
+
+from unittest import TestCase
+
+import os
+import radical.utils as ru
+import radical.pilot as rp
+
+from   radical.pilot.utils.component import Component
+
+
+try:
+    import mock
+except ImportError:
+    from unittest import mock
+
+
+# ------------------------------------------------------------------------------
+#
+class TestComponent(TestCase):
+
+    # --------------------------------------------------------------------------
+    #
+    @mock.patch.object(Component, '__init__', return_value=None)
+    def test_output(self, mocked_init):
+
+        component = Component(None, None)
+
+        component._outputs = {'test_state':[]}
+
+        with self.assertRaises(ValueError):
+            component.output(things='test_thing')
+
+        with self.assertRaises(ValueError):
+            component.output(things='test_thing', state='test_state2')
+
+        component.output(things=None,  state='test_state')
```

Comment:
How do you expect `component._output` to look like after executing this line? After every call, an assert can be evaluated.

For example:
```
expected_output = ...
self.assertEqual(component._output, expected_output)
```

Reply:
`_outputs` is a list of ZMQ endpoints.  The endpoints are stateless and don't change after the send.  We can't really check the entities unless we actually create ZMQ endpoints, so I think this is ok as is?

Reply:
It makes sense.